spec/jobs/create_group_assignment_repos_job_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe CreateGroupAssignmentReposJob, type: :job do
+  let(:organization) { GitHubFactory.create_owner_classroom_org }
+
+  let(:repo_access) do
+    student = GitHubFactory.create_classroom_student
+    RepoAccess.create(user: student, organization: organization)
+  end
+
+  let(:group_assignment) { create(:group_assignment, organization: organization, starter_code_repo_id: 40_140_589) }
+
+  after(:each) do
+    RepoAccess.destroy_all
+    Group.destroy_all
+    GroupAssignmentRepo.destroy_all
+  end
+
+  context 'a GroupAssignment with a new Grouping' do
+    it 'does not create any GroupAssignmentRepos' do
+      assert_performed_with(job: CreateGroupAssignmentReposJob, args: [group_assignment.id]) do
+        CreateGroupAssignmentReposJob.perform_later(group_assignment.id)
+      end
+
+      expect(GroupAssignmentRepo.all.count).to eql(0)
+    end
+  end
+
+  context 'a GroupAssignment with an existing Grouping with Groups', :vcr do
```

Comment:
So this tests passes locally if the cassette is not already written, otherwise the test fails.